### PR TITLE
Next.js Docs

### PR DESCRIPTION
**Pull request by deltaepsilon**
## Summary

Done
- [x] ISR support + guide
- [x] Next.js Guide — first draft, not added to highlight.io docs

Future
- [ ] Vercel Log Drain instructions
- [ ] Edge functions

## How did you test this change?

Local testing of production build.

## Are there any deployment considerations?

This is purely additive. We're bumping `@highlight-run/next` to v3.0.0.

### CODE REVIEW

**Comment by deltaepsilon** [comment]
Also, what's going on with prettier? I ran `yarn format:all` and I'm getting a ton of ticky-tacky formatting changes.

**Review by Vadman97 — APPROVED** [review]
nice! looks good so far. the setup experience definitely seems simpler. we would also want to add the README contents  to `quickstart/frontend/nextjs.tsx ` and update other docs references to the old next.js setup

File: package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
 	"resolutions": {
 		"@sideway/formula": "3.0.1",
 		"@types/dom-webcodecs": "0.1.4",
+		"ansi-color": "file:./forks/ansi-color",
```
[the remark this answers is not in the document]
Comment (deltaepsilon) [reply package.json:87]: `ansi-color` is ancient. This PR updates a bunch of OTEL stuff, and they decided to start relying on `ansi-color`... which is throwing errors on my system. Fortunately, `ansi-color` is a tiny package, so I patched it locally.

File: sdk/highlight-next/package.json
[the remark this answers is not in the document]
Comment (deltaepsilon) [reply sdk/highlight-next/package.json:1]: Will do.

File: package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
 		"dev:redis": "redis-server",
 		"docker:frontend": "yarn turbo run dev --filter frontend --filter highlight.run --filter @highlight-run/client",
 		"e2e:nestjs": "yarn turbo run dev --filter nestjs",
-		"e2e:nextjs": "yarn turbo run dev --filter nextjs...",
+		"e2e:nextjs": "yarn turbo run dev --filter nextjs",
```
[the remark this answers is not in the document]
Comment (deltaepsilon) [reply package.json:46]: Good point. Fixing...

**Comment by deltaepsilon** [comment]
> nice! looks good so far. the setup experience definitely seems simpler. we would also want to add the README contents to `quickstart/frontend/nextjs.tsx ` and update other docs references to the old next.js setup

I just took a crack at that. We can continue to iterate on these docs.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (deltaepsilon) [reply package.json:87]: We will likely be able to get rid of this patch at some point. They should realize that `ansi-color` is an iffy package.

**Pull request merged**